File: 0412_fizz_buzz/fizz_buzz.java
class Solution {
    // Time: O(n)
    // Space: O(n)
    public List<String> fizzBuzz(int n) {
        List<String> res = new ArrayList<String>();
        LinkedHashMap<Integer, String> hashMap = new LinkedHashMap<Integer, String>() {
            {
                put(3, "Fizz");
                put(5, "Buzz");
            }
        };
        for (int i = 1; i <= n; i++) {
            String tmp = "";
            for (Integer key : hashMap.keySet()) {
                if (i % key == 0) {
                    tmp += hashMap.get(key);
                }
            };
            if (tmp == "") {
                tmp += Integer.toString(i);
            }
            res.add(tmp);
        }
        return res;
    }
}

File: 0412_fizz_buzz/fizzbuzz.js
const map = new Map([
  [3, 'Fizz'],
  [5, 'Buzz'],
])

const fizzBuzz = (end = 20) => {
  const start = 1
  const res = []
  for (let i = start; i <= end; i++) {
    let tmp = ''
    map.forEach((val, key) => {
      if (i > 0 && i % key === 0) {
        tmp += val
      }
    })
    tmp = tmp || i
    res.push(tmp.toString())
  }
  return res
}
